File: sol/solution/0001-0100/0036. Valid Sudoku/Solution.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        int[] rows = new int[9], cols = new int[9], boxes = new int[9];
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                char c = board[i][j];
                if (c == '.') continue;
                int v = c - '1';
                int mask = 1 << v;
                int b = (i / 3) * 3 + (j / 3);
                if ((rows[i] & mask) != 0) return false;
                if ((cols[j] & mask) != 0) return false;
                if ((boxes[b] & mask) != 0) return false;
                rows[i] |= mask;
                cols[j] |= mask;
                boxes[b] |= mask;
            }
        }
        return true;
    }
}

File: sol/solution/0001-0100/0036. Valid Sudoku/Solution.rs
impl Solution {
    pub fn is_valid_sudoku(board: Vec<Vec<char>>) -> bool {
        let mut rows = [0u16; 9];
        let mut cols = [0u16; 9];
        let mut boxes = [0u16; 9];
        for i in 0..9 {
            for j in 0..9 {
                let c = board[i][j];
                if c == '.' { continue; }
                let v = (c as u8 - b'1') as usize;
                let mask = 1 << v;
                let b = (i / 3) * 3 + (j / 3);
                if rows[i] & mask != 0 { return false; }
                if cols[j] & mask != 0 { return false; }
                if boxes[b] & mask != 0 { return false; }
                rows[i] |= mask;
                cols[j] |= mask;
                boxes[b] |= mask;
            }
        }
        true
    }
}